info.go

```diff
@@ -388,29 +396,23 @@ func newProgramInfoFromFd(fd *sys.FD) (*ProgramInfo, error) {
 	return &pi, nil
 }
 
-func newProgramInfoFromProc(fd *sys.FD) (*ProgramInfo, error) {
-	var info ProgramInfo
-	var progType uint32
+func newProgramInfoFromProc(fd *sys.FD, pi *ProgramInfo) error {
```

Comment:
As in https://github.com/cilium/ebpf/commit/21be59580a51a37397d2041c46d33c7c529b35b8, this should be named `readProgramInfoFromProc()`.